### PR TITLE
Add option to specify default headers to be used with ChiselClient.

### PR DESCRIPTION
**Pull request by BearLemma**

### CODE REVIEW

File: cli/src/cmd/generate_src/client_lib.ts [hunk 1]
```diff
@@ -1,3 +1,28 @@
+export type ClientParams = {
```
Comment (penberg) [inline cli/src/cmd/generate_src/client_lib.ts:1]: Some JSDocs/TSDocs would be useful, I think. But maybe you want to work with @CodingDoug on that?

File: cli/src/cmd/generate_src/client_lib.ts [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+    headers?: Headers | Record<string, string>;
+};
+
+export type InternalClientParams = {
+    version?: string;
+    headers?: Headers;
+};
+
+export function cliParamsToInternal(
+    cliParams?: ClientParams,
```
Comment (penberg) [inline cli/src/cmd/generate_src/client_lib.ts:12]: You managed to confuse me with `cli` which 99.999% of the time means command line interface. If you're a lazy typist and can't be bothered to write 'clientParams', just call it `params` or `p` :)

**Review by penberg — APPROVED** [review]
lgtm, some minor stylistic comments added

File: cli/src/cmd/generate_src/client_lib.ts (hunk 1 above)
Comment (CodingDoug) [reply cli/src/cmd/generate_src/client_lib.ts:1]: Yes, we can definitely add some TSdoc to all the exported classes and type. Let's do that in another PR.

**Pull request merged**